lib/matplotlib/patheffects.py

```diff
@@ -373,3 +375,156 @@ def draw_path(self, renderer, gc, tpath, affine, rgbFace):
         if clip_path:
             self.patch.set_clip_path(*clip_path)
         self.patch.draw(renderer)
+
+
+class TickedStroke(AbstractPathEffect):
+    """
+    A line-based PathEffect which draws a path with a ticked style.
+
+    This line style is frequently used to represent constraints in
+    optimization.  The ticks may be used to indicate that one side
+    of the line is invalid or to represent a closed boundary of a
+    domain (i.e. a wall or the edge of a pipe).
+
+    The spacing, length, and angle of ticks can be controlled.
+
+    This line style is sometimes referred to as a hatched line.
+
+    See also the :doc:`contour demo example
+    </gallery/lines_bars_and_markers/lines_with_ticks_demo>`.
+
+    See also the :doc:`contours in optimization example
+    </gallery/images_contours_and_fields/contours_in_optimization_demo>`.
+    """
+
+    def __init__(self, offset=(0, 0),
+                 spacing=10.0, angle=45.0, length=np.sqrt(2),
+                 flip=False, **kwargs):
+        """
+        Parameters
+        ----------
+        offset : pair of floats, default: (0, 0)
+            The offset to apply to the path, in points.
+        spacing : float, default: 10.0
+            The spacing between ticks in points.
+        angle : float, default: 45.0
+            The angle between the path and the tick in degrees.  The angle
+            is measured as if you were an ant walking along the curve, with
+            zero degrees pointing directly ahead, 90 to your left, 180
+            behind you, and 270 to your right.
+        length : float, default: 1.414
+            The length of the tick relative to spacing.
+            Recommend length = 1.414 (sqrt(2)) when angle=45, length=1.0
+            when angle=90 and length=2.0 when angle=60.
+        flip : bool, default: False
```

Comment:
Isn't this redundant with `angle`?

`angle=45, flip=True` = `angle=315` = `angle=-45`

or similar.

Reply:
Yes it is.  In my past life (an old Java version of this code), there was a use case for flipping which side was marked without thinking about the angle.  

I don't think it hurts anything, but it can certainly be removed if the simplification is desired.

Reply:
I'd rather leave this out to keep the API footprint small and without redundancy.

You can nicely describe the angle via the range -180°:180°. Positive angles are to the left, negative ones to the right. In this picture  flip would be nothing else than changing the sign.

Reply:
Done in 3afba07ed0a.